include/RAJA/policy/vector/register.hpp

```diff
@@ -0,0 +1,87 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file containing RAJA simd policy definitions.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_policy_vector_register_HPP
+#define RAJA_policy_vector_register_HPP
+
+#include "RAJA/config.hpp"
+#include<RAJA/pattern/register.hpp>
+#include<RAJA/policy/vector/policy.hpp>
+
+
+#ifdef __AVX2__
+#include<RAJA/policy/vector/register/avx2.hpp>
+#ifndef RAJA_VECTOR_REGISTER_TYPE
+#define RAJA_VECTOR_REGISTER_TYPE RAJA::vector_avx2_register
+#endif
+#endif
+
+
+#ifdef __AVX__
+#include<RAJA/policy/vector/register/avx.hpp>
+#ifndef RAJA_VECTOR_REGISTER_TYPE
+#define RAJA_VECTOR_REGISTER_TYPE RAJA::vector_avx_register
+#endif
+#endif
+
+
+
+#ifdef RAJA_ALTIVEC
+#include<RAJA/policy/vector/register/altivec.hpp>
+#ifndef RAJA_VECTOR_REGISTER_TYPE
+#define RAJA_VECTOR_REGISTER_TYPE RAJA::vector_altivec_register
+#endif
+#endif
+
+
+// The scalar register is always supported (doesn't require any SIMD/SIMT)
+#include<RAJA/policy/vector/register/scalar.hpp>
+#ifndef RAJA_VECTOR_REGISTER_TYPE
+#define RAJA_VECTOR_REGISTER_TYPE RAJA::vector_scalar_register
+#endif
+
+
+namespace RAJA
+{
+namespace policy
+{
+  namespace vector
+  {
+
+    // This sets the default SIMD register that will be used
+    // Individual registers can
+    using vector_register_type = RAJA_VECTOR_REGISTER_TYPE;
+  }
+}
+
+
+
+  using policy::vector::vector_register_type;
+
+  template<typename T, size_t UNROLL = 1>
```

Comment:
We should add a defaulted third template argument for these wrappers (defaulted to the vector register type determined above) so that users can apply different SIMD ISAs in their code with out having to use the *VectorExt interface.  Then, we should move the 'Ext' stuff into an internal namespace.